### PR TITLE
Add diss command functionality

### PR DESCRIPTION
**Pull request by sdevalapurkar**
# Overview

- Add diss command functionality
- Ability to tag a user and diss them
- also supports the ability to pass in just any custom string for extensibility (in case you want to use a nickname or dont want to tag the user directly)

Screen Shot 2022-10-14 at 11 30 15 PM

### CODE REVIEW

File: bot/general/values.py [hunk 1]
```diff
@@ -35,6 +35,21 @@
 /status \-\- Lists the status of all status users within a group chat or private chat
 """
 
+disses = [
+    "I'mmmmmmmmmmm sooorrryyyyyyyyy for your loss",
+    "It looks like your face caught on fire and someone tried to put it out with a fork",
+    "Why don't you go play a game that is better suited for your skills like ludo",
```
Comment (AhmedNSidd) [inline bot/general/values.py:13]: hahaha i love it

File: bot/general/values.py [hunk 2]
```diff
@@ -35,6 +35,21 @@
...
+    "Your gene pool could use a little chlorine",
+    "Why play so hard to get when you're already so hard to want",
+    "I could get on your level but I don't like being on my knees as much as you do",
+    "If you ran like your mouth you'd be in really good shape",
+    "What's the difference between you and eggs? Eggs get laid and you don't",
+    "To be fair, you do have to have a very high IQ to meditate",
+    "The only thing falling faster than the GME stock are your grades",
+    "I don't watch tiktok. I'm not a virgin",
```
Comment (AhmedNSidd) [inline bot/general/values.py]: you can probs remove/reword this as well

File: bot/general/values.py [hunk 3]
```diff
@@ -35,6 +35,21 @@
...
+    "The Great Way is not difficult for those who have no preferences. When love and hate are both absent, everything becomes clear and undisguised. Make the smallest distinction, however, and heaven and earth are set infinitely apart. If you wish to see the truth, then hold no opinions for, or against, anything"
```
Comment (AhmedNSidd) [inline bot/general/values.py]: this was an inside joke as well, probs remove it too

File: bot/general/values.py [hunk 4]
```diff
@@ -35,6 +35,21 @@
...
+disses = [
+    "I'mmmmmmmmmmm sooorrryyyyyyyyy for your loss",
+    "It looks like your face caught on fire and someone tried to put it out with a fork",
+    "Why don't you go play a game that is better suited for your skills like ludo",
+    "Your gene pool could use a little chlorine",
+    "Why play so hard to get when you're already so hard to want",
+    "I could get on your level but I don't like being on my knees as much as you do",
+    "If you ran like your mouth you'd be in really good shape",
+    "What's the difference between you and eggs? Eggs get laid and you don't",
+    "To be fair, you do have to have a very high IQ to meditate",
```
Comment (AhmedNSidd) [inline bot/general/values.py:19]: There are some disses that were just inside jokes so they probably don't make sense for the general user base of this bot, you can reword them if you'd like to make them more of a diss, but it's up to you

File: bot/handlers/basic.py [hunk 5]
```diff
@@ -65,6 +66,14 @@ def about(update, context):
         disable_web_page_preview=True
     )
 
+def diss(update, context):
+    """
+    Outputs a diss message targetting the user mentioned when the command was issued
+    """
+    targetted_user = update.message.text.split(" ", 1)[1]
```
Comment (AhmedNSidd) [inline bot/handlers/basic.py]: There are a few different ways you can implement this diss functionality. Here are your options:

1. You can just diss someone by doing typing just their name as the command argument. This is nice in the sense that it makes things generic, but it's missing that "oomfh" factor to a diss.
2. You can diss someone by doing `/diss @Ahmed` and then the bot will be like
> @Ahmed You suck
This has more of an oomf factor since they are being tagged out and they would even get a notification on their mobiles.

I think you should pursue the 2nd option, personally. 

The question then arises, how do you get the tag of a user through a message and how do you tag that user? Here's two tips that will help you figure that out:

1. Try doing `/diss ` and `/diss @replace_this_with_the_username_of_your_testing_bot` and logging the update object. Try to find the difference. What additional information is the update object giving you when you tag someone in a command? 
2. Check out the `get_one_mention` function in the ./bot/handlers/common.py file. That function, or even the logic that that function uses may be helpful to you. 

Try to get it working so that `/diss @yourbot` ends up dissing your bot. Then after that, there will be a next step and some caveats to address that I will tell you about in my next review after you've done that, but this is long enough as is.

File: bot/handlers/basic.py [hunk 6]
```diff
@@ -65,6 +66,14 @@ def about(update, context):
...
+    msg = f"{values.RIGHT_POINTING_EMOJI} {targetted_user}, {random.choice(values.disses)} {values.LEFT_POINTING_EMOJI}"
```
Comment (AhmedNSidd) [inline bot/handlers/basic.py]: I don't know if I love the right and left emojis here. something simple like

`@gamerguy your mum is a whore` is probably preferable formatting here

**Review by AhmedNSidd — CHANGES_REQUESTED** [review]

File: bot/general/values.py (hunk 2 above)
Comment (sdevalapurkar) [reply bot/general/values.py]: I think it is a funny diss as it currently stands

File: bot/general/values.py (hunk 4 above)
Comment (sdevalapurkar) [reply bot/general/values.py:19]: once again, for this one I think it makes sense to the general audience too so prefer keeping it as is

**Comment by Ahmed** [comment]
Hey, could you stop pinging me?

**Comment by Ahmed** [comment]
Few things:
1-Please make sure you don't ping random users that are not related to the
PR. @Ahmed username is owned by me.
2-Write some proper unittests for this. The tests you have are not written
correctly.
3-If I'm a hiring manager from another company and see you implemented a
diss feature, it may turn me off, I think it's better to take your repo
private.

**Comment by sdevalapurkar** [comment]
@Ahmed not sure how or why you got tagged... regardless, won't tag you anymore

**Comment by AhmedNSidd** [comment]
> Few things: 1-Please make sure you don't ping random users that are not related to the PR. @Ahmed username is owned by me. 2-Write some proper unittests for this. The tests you have are not written correctly. 3-If I'm a hiring manager from another company and see you implemented a diss feature, it may turn me off, I think it's better to take your repo private.
> […](#)

1. Apologies for accidentally pinging you. I'll try to ensure that doesn't happen again.
2. We are cognizant of that and intend on adding proper testing soon. Appreciate the input.
3. If a hiring manager doesn't understand the value of having a feature that may contain profanity or a word like "diss" in it, and if I end up losing on a position based solely on that, then so be it 🤷

Appreciate the advice all in all though.

File: README.md [hunk 7]
```diff
@@ -27,6 +27,8 @@ The bot can be found at https://t.me/GamersSocialManagerBot
 /about -- The bot will reply with general information about itself.
 
 /start -- Used to start up the bot.
+
+/diss [username with or without @] -- Used to issue a random diss to the mentioned user.
```
Comment (AhmedNSidd) [inline README.md:31]: If you're doing this, it might be nice for you to add this to the /help command as well (probably best to add in the "memes" section of the help command).

**Review by AhmedNSidd — CHANGES_REQUESTED** [review]
I think just do the one thing that I commented on, and we should be good to merge this

File: README.md (hunk 7 above)
Comment (sdevalapurkar) [reply README.md:31]: Screen Shot 2022-10-20 at 8 53 27 PM
